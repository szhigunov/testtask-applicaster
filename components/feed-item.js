import React, {useCallback} from 'react';
import {
  StyleSheet,
  Text,
  View,
  Image,
  Button,
  TouchableNativeFeedback,
} from 'react-native';

const getImageUri = item =>
  item.media_group.find(({media_item}) => media_item[0].key === 'image_base')
    .media_item[0].src;
const getVideoUri = item => item.content.src;
const getHref = item => item.link.href;

export const FeedItem = ({onPress, item}) => {
  let type = item.type.value;

  const openLink = useCallback(() => {
    onPress('WebView', {
      name: item.title,
      data: {
        href: getHref(item),
      },
    });
  }, [item, onPress]);

  const openVideo = useCallback(() => {
    onPress('Video', {
      name: item.title,
      data: {
        uri: getVideoUri(item),
      },
    });
  }, [item, onPress]);

  return (
    <View style={styles.feedItem}>
      <TouchableNativeFeedback
        onPress={() => {
          if (type === 'video') {
            openVideo();
          } else {
            openLink();
          }
        }}>
        <View>
          <Text style={styles.title}>{item.title}</Text>
          <Image style={styles.image} source={{uri: getImageUri(item)}} />
          <Text style={styles.summary}>{item.summary}</Text>
          {type === 'video' && <Button title="PLAY" onPress={openVideo} />}
        </View>
      </TouchableNativeFeedback>
    </View>
  );
};

const styles = StyleSheet.create({
  feedItem: {backgroundColor: '#fff', margin: 10},
  title: {
    writingDirection: 'rtl',
    fontWeight: '600',
    fontSize: 16,
    padding: 10,
  },
  summary: {
    writingDirection: 'rtl',
    fontSize: 12,
    padding: 10,
    paddingLeft: 15,
    paddingRight: 15,
  },
  image: {
    width: '100%',
    height: 200,
  },
});
