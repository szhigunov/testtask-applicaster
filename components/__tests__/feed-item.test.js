/**
 * @format
 */

import 'react-native';
import React from 'react';
import {FeedItem} from '../feed-item';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const DATA = [
  {
    type: {
      value: 'link',
    },
    title: 'נתניהו בראיון: "מאשימים אותי בבגידה? את כולם אני אתבע"',
    summary:
      'ראש הממשלה סבור כי הרמטכ"לים לשעבר של \'כחול לבן\' יודעים שלא קיבל כסף ממכירת צוללות ומשיב אש: "אותי אתם מאשימים בבגידה? את מי שנלחם נגד כל העולם כדי למנוע מאיראן את ההסכם שאתם תמכתם בו?" | הראיון המלא הערב בחדשות 20',
    id: '343297',
    published: '',
    updated: '',
    content: {
      content:
        'ראש הממשלה סבור כי הרמטכ"לים לשעבר של \'כחול לבן\' יודעים שלא קיבל כסף ממכירת צוללות ומשיב אש: "אותי אתם מאשימים בבגידה? את מי שנלחם נגד כל העולם כדי למנוע מאיראן את ההסכם שאתם תמכתם בו?" | הראיון המלא הערב בחדשות 20',
      type: 'html',
    },
    link: {
      rel: 'alternate',
      type: 'text/html',
      href:
        'https://www.20il.co.il/mobile-post/?id=343297&native=true&shareable=true&showcontext=true&title=%D7%90%D7%A7%D7%98%D7%95%D7%90%D7%9C%D7%99%D7%94%20%D7%95%D7%97%D7%93%D7%A9%D7%95%D7%AA',
    },
    media_group: [
      {
        type: 'image',
        media_item: [
          {
            src:
              'https://www.20il.co.il/wp-content/uploads/2019/03/%D7%A0%D7%AA%D7%A0%D7%99%D7%94%D7%95-%D7%91%D7%A8%D7%90%D7%99%D7%95%D7%9F-1270x714.jpg',
            type: 'image',
            scale: 'small',
            key: 'image_base',
          },
          {
            src:
              'https://www.20il.co.il/wp-content/uploads/2019/03/%D7%A0%D7%AA%D7%A0%D7%99%D7%94%D7%95-%D7%91%D7%A8%D7%90%D7%99%D7%95%D7%9F-1270x714.jpg',
            type: 'image',
            scale: 'large',
            key: 'image_base_2',
          },
        ],
      },
    ],
    author: {
      name: '',
    },
  },
];

describe('Component/FeedItem', () => {
  it('FeedItem renders correctly', () => {
    renderer.create(<FeedItem item={DATA[0]} />);
  });
});
