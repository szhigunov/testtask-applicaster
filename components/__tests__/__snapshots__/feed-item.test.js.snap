// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/FeedItem FeedItem renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "margin": 10,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "600",
          "padding": 10,
          "writingDirection": "rtl",
        }
      }
    >
      נתניהו בראיון: "מאשימים אותי בבגידה? את כולם אני אתבע"
    </Text>
    <Image
      source={
        Object {
          "uri": "https://www.20il.co.il/wp-content/uploads/2019/03/%D7%A0%D7%AA%D7%A0%D7%99%D7%94%D7%95-%D7%91%D7%A8%D7%90%D7%99%D7%95%D7%9F-1270x714.jpg",
        }
      }
      style={
        Object {
          "height": 200,
          "width": "100%",
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 12,
          "padding": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "writingDirection": "rtl",
        }
      }
    >
      ראש הממשלה סבור כי הרמטכ"לים לשעבר של 'כחול לבן' יודעים שלא קיבל כסף ממכירת צוללות ומשיב אש: "אותי אתם מאשימים בבגידה? את מי שנלחם נגד כל העולם כדי למנוע מאיראן את ההסכם שאתם תמכתם בו?" | הראיון המלא הערב בחדשות 20
    </Text>
  </View>
</View>
`;
